webofneeds/won-node/src/main/java/won/node/camel/processor/general/ReferenceToUnreferencedMessageAddingProcessor.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2012  Research Studios Austria Forschungsges.m.b.H.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package won.node.camel.processor.general;
+
+import com.hp.hpl.jena.query.Dataset;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import won.cryptography.rdfsign.SigningStage;
+import won.protocol.message.SignatureReference;
+import won.protocol.message.WonMessage;
+import won.protocol.message.WonMessageType;
+import won.protocol.message.processor.WonMessageProcessor;
+import won.protocol.message.processor.exception.WonMessageProcessingException;
+import won.protocol.message.processor.impl.WonMessageSignerVerifier;
+import won.protocol.model.DatasetHolder;
+import won.protocol.model.MessageEventPlaceholder;
+import won.protocol.repository.DatasetHolderRepository;
+import won.protocol.repository.MessageEventRepository;
+
+import java.net.URI;
+import java.util.List;
+
+import static won.protocol.message.WonMessageType.CONNECT;
+import static won.protocol.message.WonMessageType.HINT_MESSAGE;
+import static won.protocol.message.WonMessageType.OPEN;
+
+/**
+ * Created by fkleedorfer on 22.07.2016.
+ */
+public class ReferenceToUnreferencedMessageAddingProcessor implements WonMessageProcessor
```

Comment:
it's a processor in camel, and we run it in a multi-threaded environment, so, yes it may happen that messages are referenced multiple times. As long as we don't miss any, I think that's ok